e2e/tests/onchain_settlement.rs

```diff
@@ -29,20 +28,23 @@ use solver::{
 use std::{collections::HashSet, str::FromStr, sync::Arc, time::Duration};
 use web3::signing::SecretKeyRef;
 
+mod ganache;
+
 const TRADER_A_PK: [u8; 32] =
     hex!("0000000000000000000000000000000000000000000000000000000000000001");
 const TRADER_B_PK: [u8; 32] =
     hex!("0000000000000000000000000000000000000000000000000000000000000002");
 
-const NODE_HOST: &str = "http://127.0.0.1:8545";
 const API_HOST: &str = "http://127.0.0.1:8080";
 const ORDER_PLACEMENT_ENDPOINT: &str = "/api/v1/orders/";
 
 #[tokio::test]
-async fn test_with_ganache() {
+async fn ganache_onchain_settlement() {
+    ganache::test(onchain_settlement).await;
```

Comment:
Just curious, could this be also done via an annotation similar to `#[tokio:test]`?

Reply:
Yes, but it would require moving this code to a `proc-macro` crate.

Reply:
Its actually quite fun :smile:. I wouldn't mind helping get someone do a PoC with a pair programming session on ecosystem-friday if anyone is interested :smile: